invenio_files_rest/models.py

```diff
@@ -732,9 +741,124 @@ def get_by_bucket(cls, bucket, versions=False):
         return cls.query.filter(*args).order_by(cls.key, cls.created.desc())
 
 
+class MultipartObject(db.Model, Timestamp):
+    """Model for storing files in chunks.
+
+    A multipart object handles the file upload in chunks.
+
+    First it creates an ``upload_id`` and a file instance and updates the file
+    until the ``finilize()`` called. Then the multipart object will mark the
+    instance as completed and create a ObjectVersion with the uploaded file.
+    """
+
+    __tablename__ = 'files_multipart_object'
+
+    bucket_id = db.Column(
+        UUIDType,
+        db.ForeignKey(Bucket.id, ondelete='RESTRICT'),
+        default=uuid.uuid4,
+        primary_key=True, )
+    """Bucket identifier."""
+
+    key = db.Column(
+        db.String(255),
+        primary_key=True, )
+    """Key identifying the object."""
+
+    upload_id = db.Column(
+        UUIDType,
+        primary_key=True,
```

Comment:
Not sure if ``upload_id`` should be part of the primary key. Probably it should just have a unique constraint. I.e. do we allow having more than one multipart upload for a given key in a bucket? It could probably be fun if multiple users upload to the same key at the same time though.